name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/editor/ClientInput.java

```diff
@@ -1,511 +1,605 @@
+            return;
+
+        StringBuilder buf = new StringBuilder();
+        for (ReportingPeriod p : reportingPeriods)
+        {
+            p.writeTo(buf);
+            buf.append(';');
+        }
+
+        getPreferenceStore().setValue(REPORTING_PERIODS_KEY, buf.toString());
+    }
+
+    @Inject
+    @Optional
+    public void onExchangeRatesLoaded(@UIEventTopic(UIConstants.Event.ExchangeRates.LOADED) Object obj)
+    {
+        if (exchangeRateProviderFacory != null)
+        {
+            exchangeRateProviderFacory.clearCache();
+            listeners.forEach(ClientInputListener::onRecalculationNeeded);
+        }
+    }
+
+    private void scheduleOnlineUpdateJobs()
+    {
+        if (preferences.getBoolean(UIConstants.Preferences.UPDATE_QUOTES_AFTER_FILE_OPEN, true))
+        {
+            Job initialQuoteUpdate = new UpdateQuotesJob(client,
+                            EnumSet.of(UpdateQuotesJob.Target.LATEST, UpdateQuotesJob.Target.HISTORIC));
+            initialQuoteUpdate.schedule(1000);
+
+            CreateInvestmentPlanTxJob checkInvestmentPlans = new CreateInvestmentPlanTxJob(client,
+                            exchangeRateProviderFacory);
+            checkInvestmentPlans.startAfter(initialQuoteUpdate);
+            checkInvestmentPlans.schedule(1100);
+
+            int thirtyMinutes = 1000 * 60 * 30;
+            Job job = new UpdateQuotesJob(client, EnumSet.of(UpdateQuotesJob.Target.LATEST)).repeatEvery(thirtyMinutes);
+            job.schedule(thirtyMinutes);
+            regularJobs.add(job);
+
+            int sixHours = 1000 * 60 * 60 * 6;
+            job = new UpdateQuotesJob(client, EnumSet.of(UpdateQuotesJob.Target.HISTORIC)).repeatEvery(sixHours);
+            job.schedule(sixHours);
+            regularJobs.add(job);
+
+            new SyncOnlineSecuritiesJob(client).schedule(2000);
+        }
+    }
+
+    private void scheduleAutoSaveJob()
+    {
+        
+        int delay = preferences.getInt(UIConstants.Preferences.AUTO_SAVE_FILE, 15);
+        IPreferenceChangeListener listener = new IPreferenceChangeListener()
+        {
+
+            @Override
+            public void preferenceChange(PreferenceChangeEvent event)
+            {
+                if (event.getKey().contentEquals(UIConstants.Preferences.AUTO_SAVE_FILE))
+                {
+
+                    for (Iterator<Job> i = regularJobs.iterator(); i.hasNext();)
+                    {
+                        Job j = i.next();
+                        if (j instanceof AutoSaveJob)
+                        {
+                            ((AutoSaveJob) j).setDelay(getAutoSavePrefs());
+                            ((AutoSaveJob) j).schedule(getAutoSavePrefs());
+                            ((AutoSaveJob) j).wakeUp(getAutoSavePrefs());
+                        }
+                    }
+                }
+            }
+        };
+        preferences.addPreferenceChangeListener(listener);
+
+        if (delay > 0)

```

Comment:
Ich denke der Job sollte *immer* angelegt werden (aber nur ge"scheduled" wenn das delay > 0 ist). Ansonsten kann ein Benutzer nicht von 0 Minuten (=kein Autosave) auf 10 Minuten (=autosave alle 10 Minuten) ändern. Der Job existiert nicht in der Liste der ```regularJobs```

Reply:
Guter Punkt. Ändere ich.